lib/spack/spack/binary_distribution.py

```diff
@@ -1588,7 +1588,7 @@ def relocate_package(spec, allow_root):
     prefix_to_prefix_text[old_prefix] = new_prefix
     prefix_to_prefix_bin[old_prefix] = new_prefix
     prefix_to_prefix_text[old_layout_root] = new_layout_root
-    prefix_to_prefix_bin[old_layout_root] = new_layout_root
+    # prefix_to_prefix_bin[old_layout_root] = new_layout_root
```

Comment:
I found this one odd too. Was it intended as a generic fallback?

Reply:
I'm honestly not sure, all the tests pass without it (though I'm still fixing up the rework for a single regex version).

Reply:
I'll put this back in place, but after the prefix loop